source/Core/Drivers/BootLogo.cpp

```diff
@@ -46,9 +55,9 @@ void BootLogo::showNewFormat(const uint8_t *ptrLogoArea) {
       osDelay(interFrameDelay * 5);
     }
     // 1024 less the header type byte and the inter-frame-delay
-    if (getSettingValue(SettingsOptions::LOGOTime) < 5 && (position == 1022 || len == 0)) {
+    if (getSettingValue(SettingsOptions::LOGOTime) > 0 && (position == 1022 || len == 0)) {
       // Delay here until button is pressed or its been the amount of seconds set by the user
-      waitForButtonPressOrTimeout(TICKS_SECOND * getSettingValue(SettingsOptions::LOGOTime));
+      delay();
       return;
     }
   } while (buttons == BUTTON_NONE);
```

Comment:
@Ralim
Good morning Sir,
### nice to hear from you again! 😃😊
## Hopefully you are doing well!

Now to the code.
Isn’t this line enough?
What does your approach do to make the code more reliable?
No accusation or sarcasm, just trying to understand. You know I’m not that good in this field.

Reply:
Hahah dont worry I've not gone anywhere; just been poking other projects for a bit of a break.

So I just added the case for if the timeout is 5; then we use the `waitForButtonPress` instead since we dont want a timeout.

Reply:
Alright! 👍

Ah, I see.
So, am I right about the do-while-loop being redundant?